libr/util/file.c

```diff
@@ -1549,11 +1549,17 @@ R_API RList* r_file_glob(const char *_globbed_path, int maxdepth) {
 		if (last_slash) {
 			glob_ptr = last_slash + 1;
 			if (globbed_path[0] == '~') {
-				char *rpath = r_str_newlen (globbed_path + 2, last_slash - globbed_path - 1);
+				char *rpath = NULL;
+				if (last_slash - globbed_path - 1) {
+					rpath = r_str_ndup (globbed_path + 2, last_slash - globbed_path - 1);
+				}
 				path = r_file_home (r_str_get (rpath));
 				free (rpath);
 			} else {
-				path = r_str_newlen (globbed_path, last_slash - globbed_path + 1);
+				path = NULL;
+				if (last_slash - globbed_path + 1) {
```

Comment:
same logic here, try to simplify the. logic to avoid repeating patterns.

* do not compute length twice
* the negative length check should be done in a macro or api (R_STR_NDUP)
* dont assign path twice, use else branch or so. the old r_str_newlen was "cleaner" imho. but not clearer